src/sentry/snuba/metrics/extraction.py

```diff
@@ -394,15 +399,18 @@ def _get_percentile_op(args: Sequence[str]) -> Optional[MetricOperationType]:
     return None
 
 
-def _get_args_support(function: str, args: Sequence[str]) -> SupportedBy:
-    if len(args) == 0:
+def _get_columns_support(
+    columns: Sequence[str], used_in_function: Optional[str] = None
+) -> SupportedBy:
+    if len(columns) == 0:
+        # This is also called to check columns in functions.
         return SupportedBy.both()
 
-    # apdex can have two variations, either apdex() or apdex(value).
-    if function == "apdex":
+    if used_in_function == "apdex":
+        # apdex can have two variations, either apdex() or apdex(value).
         return SupportedBy(on_demand_metrics=True, standard_metrics=False)
 
-    arg = args[0]
+    arg = columns[0]
```

Comment:
Refactored, what it's really doing is the same as checking whether the arg in a function is compatible, which basically checks whether a 'column' is compatible since functions take a column as a arg in some cases.

I did it for parity here but looking at this again I think `columns` should all be checked for support, and the result joined. Not sure why we're getting the first index only here.

Reply:
For clarity I think I'm re-arranging this so the aggregate args and groupbys are checked separately.